python/python/bystro/proteomics/handler.py

```diff
@@ -0,0 +1,362 @@
+    if not isinstance(field_path, list) or data_for_end_of_path is None:
+        return data_for_end_of_path
+    assert isinstance(data_for_end_of_path, dict)  # noqa: S101
+    for child_field in field_path:
+        data_for_end_of_path = data_for_end_of_path.get(child_field)
+        if data_for_end_of_path is None:
+            return data_for_end_of_path
+
+    return data_for_end_of_path
+
+
+Sub = list[str | None] | str | None
+PositionData = list[Sub]
+Column = None | list[PositionData]
+Row = list[Column]
+
+
+def _make_output_string(rows: list[Row], delims: dict[str, str]) -> bytes:
+    output_rows = [_do_row(row, delims) for row in rows]
+    return bytes("\n".join(output_rows) + "\n", encoding="utf-8")
+
+
+def _do_row(row: Row, delims: dict[str, str]) -> str:
+    output_row = [_do_column(col, delims) for col in row]
+    return delims["field"].join(output_row)
+
+
+def _do_column(column: Column, delims: dict[str, str]) -> str:
+    if column is None:
+        return delims["empty_field"]
+    if not isinstance(column, list):
+        return str(column)
+    output_column = [
+        _do_position_data(pos_data, delims) for pos_data in column if isinstance(pos_data, list)
+    ]
+    return delims["overlap"].join(output_column)
+
+
+def _do_position_data(position_data: list[Sub], delims: dict[str, str]) -> str:
+    # if position_data is None:
+    inner_values = [_do_sub(sub, delims) for sub in position_data]
+    return delims["position"].join(inner_values)
+
+
+def _do_sub(sub: Sub, delims: dict[str, str]) -> str:
+    if sub is None:
+        return delims["empty_field"]
+    if isinstance(sub, list):
+        sub_strings = [_to_string_or_empty_field_char(s, delims) for s in sub]
+        return delims["value"].join(sub_strings)
+    elif isinstance(sub, str):
+        return sub
+    else:
+        raise TypeError(sub)
+        return str(sub)
+
+
+def _to_string_or_empty_field_char(x: str | None, delims: dict[str, str]) -> str:
+    return str(x) if x is not None else delims["empty_field"]
+
+
+from typing import cast
+
+
+def _process_rows(resp: dict, field_names: list[str]) -> list[Row]:
+    parent_fields, child_fields = _get_header(field_names)
+    try:
+        discordant_idx = field_names.index("discordant")
+    except ValueError as val_err:
+        err_msg = f"response: {resp} with field_names: {field_names} lacked field: 'discordant'"
+        raise ValueError(err_msg) from val_err
+    rows: list[Row] = []
+    docs = resp["hits"]["hits"]
+    for doc in docs:
+        row = []
+        for parent_field, child_field in zip(parent_fields, child_fields):
+            row.append(_populate_data(child_field, doc["_source"].get(parent_field)))
+        row[discordant_idx][0][0] = cast(str, _fix_discordancy(row[discordant_idx][0][0]))  # type: ignore
+        rows.append(row)  # type: ignore
+    return rows
+
+
+def _fix_discordancy(x: bool | str) -> str:
+    return {False: "0", True: "1", "0": "0", "1": "1"}[x]
```

Comment:
Why are we trying to fix discordant fields?

Reply:
These are just rows that the user submitted with ref != assembly ref. If there are a lot of these they used the wrong genome, else they should be careful with the data. We should never fix these guys